dashboard/app/helpers/levels_helper.rb

```diff
@@ -179,15 +179,22 @@ def app_options
     # Unsafe to generate these twice, so use the cached version if it exists.
     return @app_options unless @app_options.nil?
 
-    if (@level.channel_backed? && params[:action] != 'edit_blocks') || @level.is_a?(Javalab)
+    should_load_channel = (@level.channel_backed? && params[:action] != 'edit_blocks') || @level.is_a?(Javalab)
+    # If we have loaded a user, the page must not be publicly cached
+    level_not_cached = current_user || @user
+    if should_load_channel && level_not_cached
       view_options(
         channel: get_channel_for(@level, @script&.id, @user),
-        server_project_level_id: @level.project_template_level.try(:id),
       )
       # readonly if viewing another user's channel
       readonly_view_options if @user
     end
 
+    view_options(
+      should_load_channel: should_load_channel,
```

Comment:
Similar to the earlier comment, I'm finding `should_load_channel` a little confusing.  I was struggling to come up with a suggestion but maybe just `channel_backed` would be a good name?